sdk-base/src/main/java/com/mapbox/maps/plugin/viewport/experimental/ViewportStatusObserver.kt

```diff
@@ -0,0 +1,15 @@
+package com.mapbox.maps.plugin.viewport.experimental
+
+/**
+ * Observer that gets notified whenever [ViewportStatus] changes.
+ */
+fun interface ViewportStatusObserver {
+  /**
+   * Called whenever [ViewportStatus] changes.
+   *
+   * @param from The previous [ViewportStatus], null if the previous [ViewportStatus] is IDLE.
+   * @param to The current [ViewportStatus], null if the current [ViewportStatus] is IDLE.
+   * @param reason The reason that the state has been changed.
+   */
+  fun onViewportStateChanged(from: ViewportStatus?, to: ViewportStatus?, reason: String)
```

Comment:
When could a status be null? Is there a way to make these non-nullable?

Reply:
I think these can be non-nullable now that Viewport.status is non-nullable.

Reply:
right, will refactor these.